onnxruntime/core/providers/webnn/builders/model.cc

```diff
@@ -48,11 +60,24 @@ Status Model::Predict(const InlinedHashMap<std::string, OnnxTensorData>& inputs,
     }
     auto num_elements = SafeInt<size_t>(Product(tensor.tensor_info.shape));
     emscripten::val view{emscripten::typed_memory_view(num_elements, static_cast<const float*>(tensor.buffer))};
-    wnn_outputs_.set(name, view);
+    val_vec.insert({name, view});
```

Comment:
`val_vec` is only useful for `SharedArrayBuffer.as<bool>()`, move it to that block.